core/js/src/main/scala/zio/internal/Mailbox.scala

```diff
@@ -0,0 +1,55 @@
+package zio.internal
+
+import zio.stacktracer.TracingImplicits.disableAutoTrace
+
+import scala.annotation.tailrec
+
+private[internal] final class Mailbox[A] extends Serializable {
+
+  private[this] var read  = new Node(null.asInstanceOf[A])
+  private[this] var write = read
+
+  @throws[NullPointerException]("if the element is null")
+  def add(data: A): Unit = {
+    if (null == data) throw new NullPointerException()
```

Comment:
redundant null check, we know all call sites and none introduces null

Reply:
Done